pisa/core/container.py

```diff
@@ -195,7 +195,7 @@ def __getitem__(self, key):
 
     def __setitem__(self, key, value):
         for container in self:
-            container[key] = np.copy(value)
```

Comment:
We need to copy, otherwise we can get strange behaviour after unlinking!

Reply:
hmm, really? Then we need to check the memory behavior again, it caused a memory leak for me, or at least the memory profiler told me that it did. But maybe that was also just because I was investigating the nusquids memory leak and it really wasn't caused by this line.

Reply:
the thing is, if we do not copy, then changing values after unlinking in one container will affect the values in another container. (because it will be referencing the same object).
I don't remember seeing a memory leak, but we can double check. I'd assume the garbage collector to get rid of of what was in there before once we overwrite (?)

Reply:
Yes I thought about that issue as well, but on the other hand, the behavior is already known from dictionaries, so people should in theory be aware that setting an item usually only passes a reference and that they need to copy if they want them to be independent... If we always copy, then we might be allocating and de-allocating large chunks of memory unnecessarily. The memory leak was probably because nusquids messed up reference counting (which is fixed now).

Reply:
Ah wait, I think I didn't fully appreciate the problem. You mean that, after unlinking, separate containers share the same referenced object and thus affect each other?

Reply:
Maybe that could be fixed by explicitly copying when we unlink?

Reply:
we could do that.....we'd need to keep track of what variables are changed and then copy when unlinking.....so you would safe some time if a key is altered more than once in the linked state......which may be rare.....
persoanlly i'd rather leave it simple. Or do you anticipate perfromance issues?